src/daemon/neightable.rs

```diff
@@ -12,8 +12,8 @@ use super::link_connection::LinkConnection;
 #[allow(clippy::type_complexity)]
 pub struct NeighTable {
     table: DashMap<Fingerprint, (LinkConnection, Option<Instant>, Immortal)>,
-    send_incoming: Sender<RawPacket>,
-    recv_incoming: Receiver<RawPacket>,
+    send_incoming: Sender<(Fingerprint, RawPacket)>,
```

Comment:
we can in fact remove these channels from NeighTable. Instead, we can get rid of the whole `peel_forward_loop`, and instead call a `peel_forward` function (refactored from the body of that loop) every time we have a packet to peel and forward, from anywhere in the function

this can be in a `peel_forward.rs` file

then, NeighTable would be a straightforward mapping of Fingerprint to Senders, used for *sending* packets but not receiving.

NeighTable would also no longer contain any tasks. The task that reads from the connection and calls `peel_forward` will, of course, be started in `per_route_tasks`.